commits.go

```diff
@@ -176,6 +176,9 @@ func (s *CommitsService) GetCommit(pid interface{}, sha string, options ...Optio
 	if err != nil {
 		return nil, nil, err
 	}
+	if sha == "" {
+		return nil, nil, fmt.Errorf("sha mustn't be empty")
+	}
```

Comment:
I appreciate what you are trying to do here, but (at least until now) this package doesn't do any input validation and instead leverages the Gitlab API validation logic. So it just forwards the request and returns any errors that the API encounters while validating. This seems more robust as there are a lot of things that are tested and validated by the Gitlab API endpoints which also change over time.

So I'm inclined to close this PR instead of merge it, unless you have a very good reason why this specific check needs to be in here?

Reply:
Actually in this case when sha is an empty string, the sdk will make a request to another [api](https://docs.gitlab.com/ee/api/commits.html#list-repository-commits) which gets a list of commits, quite confusing

Reply:
I think we can merge this pr